main.go

```diff
@@ -157,8 +164,12 @@ func (c *infomaniakDNSProviderSolver) do(ch *v1alpha1.ChallengeRequest, action a
 		return err
 	}
 
-	if strings.HasSuffix(source, domain.CustomerName) {
-		source = source[:len(source)-len(domain.CustomerName)-1]
+	domainASCII, err := domain.ASCIIName()
+	if err != nil {
+		return err
+	}
+	if strings.HasSuffix(source, domainASCII) {
+		source = source[:len(source)-len(domainASCII)-1]
 	}
```

Comment:
If this does what I think it does (remove the prefix and the trailing dot), we can use this to make it clear :

```suggestion
	if strings.HasSuffix(source, domainASCII) {
		source = strings.TrimSuffix(source, domainASCII)
		source = strings.TrimSuffix(source, ".")
	}
```

Reply:
Fixed, thx :+1: